### PR TITLE
fix: expanded view not working on shared view and shared bases

### PR DESCRIPTION
**Pull request by DarkPhoenix2704**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Comment by dstala** [comment]
@DarkPhoenix2704 please check coderabbit review comments

**Review by dstala — CHANGES_REQUESTED** [review]
- [ ] expanded record for Kanban, Save record button is enabled even when we have not made changes. needs a fix.
- [x] calendar - wasn't able to expand record from shared view
- [x] code rabbit review comments
- [x] calendar shared base issue after expanding record
https://github.com/nocodb/nocodb/assets/86527202/5e7bee27-fc06-4273-86f8-bdfcab05b058

File: packages/nc-gui/composables/useExpandedFormStore.ts [hunk 1]
```diff
@@ -349,53 +350,62 @@ const [useProvideExpandedFormStore, useExpandedFormStore] = useInjectionState((m
...
+        rowMeta: {},
+      })
+    } catch (err: any) {
+      if (err.response?.status === 404) {
+        const router = useRouter()
+        message.error(t('msg.noRecordFound'))
+        router.replace({ query: {} })
+      } else {
+        message.error(`${await extractSdkResponseErrorMsg(err)}`)
+      }
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nc-gui/composables/useExpandedFormStore.ts:407]: They are dependent, hence required to be in a single try catch

File: packages/nc-gui/components/smartsheet/calendar/MonthView.vue [hunk 2]
```diff
@@ -567,11 +567,12 @@ const stopDrag = (event: MouseEvent) => {
 }
 
 const dragStart = (event: MouseEvent, record: Row) => {
-  if (!isUIAllowed('dataEdit') || resizeInProgress.value || !record.rowMeta.id) return
+  if (resizeInProgress.value || !record.rowMeta.id) return
   let target = event.target as HTMLElement
   isDragging.value = false
 
   dragTimeout.value = setTimeout(() => {
+    if (!isUIAllowed('dataEdit')) return
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nc-gui/components/smartsheet/calendar/MonthView.vue:575]: The dragStart is triggered from a HTML Element. Hence no need to null check

File: packages/nc-gui/components/smartsheet/calendar/DayView/DateTimeField.vue [hunk 3]
```diff
@@ -681,13 +681,13 @@ const stopDrag = (event: MouseEvent) => {
...
 
 const dragStart = (event: MouseEvent, record: Row) => {
-  if (!isUIAllowed('dataEdit')) return
   let target = event.target as HTMLElement
 
   isDragging.value = false
 
   // We use a timeout to determine if the user is dragging or clicking on the record
   dragTimeout.value = setTimeout(() => {
+    if (!isUIAllowed('dataEdit')) return
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nc-gui/components/smartsheet/calendar/DayView/DateTimeField.vue:690]: The eventListeneers are remove on setTimeout.

**Review by dstala — APPROVED** [review]

**Pull request merged**